user.go

```diff
@@ -5,15 +5,33 @@ import "gopkg.in/mgo.v2/bson"
 type User struct {
 	Id       bson.ObjectId `json:"id" bson:"_id,omitempty"`
 	Username string        `json:"username" bson:"username"`
+	FullName string 	`json:"fullname" bson:"fullname"`
+	Email	 string `json:"email" bson:"email"`
+	PasswordHash string `json:"password" bson:"password"`
+	AccessToken string `json:"accessToken" bson:"accessToken"`
 	Courses  []string      `json:"courses" bson:"courses"`
 }
 
 type UserRepository interface {
 	Store(user *User) error
 	Find(id string) (error, *User)
+	FindByUsername(username string) (error, *User)
+	FindByAccessToken(accessToken string) (error, *User)
+	UpdateAccessToken(user *User) (error, *User)
+}
+
+type UserAuthenticator interface {
```

Comment:
As mentioned above, I would consider these part of the userService, so they can be either removed or moved to the `UserService` interface. One thing that might be worth thinking about is testability in case of the latter. In that case the mocks (#15) would need to implement the `auth` functions in order to implement this interface as well. The question is though, if they are needed outside the userService. I don't see a case where that is needed yet, so I think I would make those internal functions of userService.

Reply:
Since some of our authentication methods are random, it would be better if we would mock those methods for testability.
Wouldn't it be the easiest if we make an own interface for authentication? Otherwise we would have to mock the whole service.